okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -13,881 +13,666 @@
- * }</pre>
+ * This example shows a call with a short 500 millisecond timeout:
+ * ```
+ * OkHttpClient eagerClient = client.newBuilder()
+ *     .readTimeout(500, TimeUnit.MILLISECONDS)
+ *     .build();
+ * Response response = eagerClient.newCall(request).execute();
+ * ```
  *
- * <h3>Shutdown isn't necessary</h3>
+ * ## Shutdown isn't necessary
  *
- * <p>The threads and connections that are held will be released automatically if they remain idle.
+ * The threads and connections that are held will be released automatically if they remain idle.
  * But if you are writing a application that needs to aggressively release unused resources you may
  * do so.
  *
- * <p>Shutdown the dispatcher's executor service with {@link ExecutorService#shutdown shutdown()}.
- * This will also cause future calls to the client to be rejected. <pre>   {@code
- *
- *     client.dispatcher().executorService().shutdown();
- * }</pre>
+ * Shutdown the dispatcher's executor service with [shutdown()][ExecutorService.shutdown].
+ * This will also cause future calls to the client to be rejected.
+ * ```
+ * client.dispatcher().executorService().shutdown();
+ * ```
  *
- * <p>Clear the connection pool with {@link ConnectionPool#evictAll() evictAll()}. Note that the
- * connection pool's daemon thread may not exit immediately. <pre>   {@code
+ * Clear the connection pool with [evictAll()][ConnectionPool.evictAll]. Note that the
+ * connection pool's daemon thread may not exit immediately.
+ * ```
+ * client.connectionPool().evictAll();
+ * ```
  *
- *     client.connectionPool().evictAll();
- * }</pre>
  *
- * <p>If your client has a cache, call {@link Cache#close close()}. Note that it is an error to
+ * If your client has a cache, call [close()][Cache.close]. Note that it is an error to
  * create calls against a cache that is closed, and doing so will cause the call to crash.
- * <pre>   {@code
+ * ```
+ * client.cache().close();
+ * ```
  *
- *     client.cache().close();
- * }</pre>
- *
- * <p>OkHttp also uses daemon threads for HTTP/2 connections. These will exit automatically if they
+ * OkHttp also uses daemon threads for HTTP/2 connections. These will exit automatically if they
  * remain idle.
  */
-public class OkHttpClient implements Cloneable, Call.Factory, WebSocket.Factory {
-  static final List<Protocol> DEFAULT_PROTOCOLS = Util.immutableList(
-      Protocol.HTTP_2, Protocol.HTTP_1_1);
+class OkHttpClient internal constructor(
+  builder: Builder
+) : Cloneable, Call.Factory, WebSocket.Factory {
 
-  static final List<ConnectionSpec> DEFAULT_CONNECTION_SPECS = Util.immutableList(
-      ConnectionSpec.MODERN_TLS, ConnectionSpec.CLEARTEXT);
+  @get:JvmName("dispatcher")
+  val dispatcher: Dispatcher = builder.dispatcher
 
-  static {
-    Internal.instance = new Internal() {
-      @Override public void addLenient(Headers.Builder builder, String line) {
-        builder.addLenient(line);
-      }
+  @get:JvmName("proxy")
+  val proxy: Proxy? = builder.proxy
```

Comment:
By making it a public val rather than a private field, you’re changing the API for Kotlin callers. I don’t think we’re ready for this. In particular I’d like to defer changing APIs for Kotlin callers until we have an automatic migration path based on strong deprecation.

Could you change this PR to make zero API changes for Java and Kotlin callers? Then later in follow-up we can make things more Kotlin friendly, and support those changes by test cases.

Reply:
Can do